packages/cipherstash-proxy/src/cli/migrate/mod.rs

```diff
@@ -0,0 +1,309 @@
+                    _ => {
+                        error!(target: MIGRATE, msg = "Invalid row. You should not even be here.");
+                        unreachable!()
+                    }
+                }
+            }
+
+            if self.verbose {
+                info!(target: MIGRATE, msg = "Encrypting", ?records);
+            }
+
+            debug!(target: MIGRATE, msg = "Update", update_sql = update_sql);
+
+            if self.commit() {
+                {
+                    client.simple_query(&update_sql).await?;
+                    client.simple_query("COMMIT;").await?;
+                }
+            }
+
+            if row_count < self.batch_size {
+                info!(
+                    target: MIGRATE,
+                    msg = "Encryption complete",
+                    updated = updated_count,
+                    batches = batch_count+1,
+                );
+
+                println!();
+                break;
+            }
+
+            batch_count += 1;
+        }
+
+        Ok(())
+    }
+}
+
+/// Parse a single key-value pair - copied from clap example https://github.com/clap-rs/clap/blob/master/examples/typed-derive.rs#L25
+fn parse_key_val<T, U>(s: &str) -> Result<(T, U), Box<dyn ClapError + Send + Sync + 'static>>
+where
+    T: std::str::FromStr,
+    T::Err: ClapError + Send + Sync + 'static,
+    U: std::str::FromStr,
+    U::Err: ClapError + Send + Sync + 'static,
+{
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid KEY=value: no `=` found in `{s}`"))?;
+    Ok((s[..pos].parse()?, s[pos + 1..].parse()?))
+}
+
+pub async fn connect_with_tls(
+    connection_string: &str,
+    with_tls_verification: bool,
+) -> Result<Client, Error> {
+    let mut tls_config = ClientConfig::with_platform_verifier();
+
+    if !with_tls_verification {
+        let mut dangerous = tls_config.dangerous();
+        dangerous.set_certificate_verifier(Arc::new(NoCertificateVerification {}));
+    }
+
+    let tls = tokio_postgres_rustls::MakeRustlsConnect::new(tls_config);
+    let (client, connection) = tokio_postgres::connect(connection_string, tls)
+        .await
+        .inspect_err(|_| {
+            error!(target: MIGRATE, msg = "Error connecting Encryption Migrator to Proxy");
+        })?;
+
+    tokio::spawn(async move {
+        if let Err(err) = connection.await {
+            error!("Connection error: {}", err);
+        }
+    });
+    Ok(client)
+}
+
+pub async fn connect_with_no_tls(connection_string: &str) -> Result<Client, Error> {
+    let (client, connection) = tokio_postgres::connect(connection_string, NoTls)
+        .await
+        .inspect_err(|_| {
+            error!(target: MIGRATE, msg = "Error connecting Encryption Migrator to Proxy");
+        })?;
+
+    tokio::spawn(async move {
+        if let Err(err) = connection.await {
+            error!("Connection error: {}", err);
+        }
+    });
+    Ok(client)
+}
+
+mod tests {
```

Comment:
Can this be under `#[cfg(test)]`, unless it's needed outside of tests?

Reply:
I did have `#[cfg(test)]` but the module wasn't available in the integration tests.
I think because it is a dependency, so the integration suite is running with `cfg=test` but the dependency isn't?
Open to other ideas. Maybe I should import and wrap in the test suite instead

Reply:
Oh I see. That makes sense. Ta. Let's leave it this way.